### DISCUSSION TITLE
How do you get the full url for srcset responsive images?

### DISCUSSION THREAD

**Discussion by Slgoetz** [spatie/laravel-medialibrary#2045] [Q&A]
I am trying to get the full URL for the image's srcset during an API call. If I do custom media conversions with specified names I can build my own srcset, but I can't seem to figure out how to return the built in responsive image feature with full URLs or srcset ready.

```  
public function registerMediaCollections(): void{
        $this->addMediaCollection('hero_image')
            ->singleFile()
            ->withResponsiveImages();
}


public function hero_image()
{
    if ($this->hasMedia('hero_image')) {
        return $this->getFirstMedia('hero_image');
    }

}


```

**Comment by garymtate** [comment]
I've just found this which might help?

`$category->getFirstMedia('category-image')->getSrcset('main')`
